core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -142,72 +283,150 @@ public TermsQueryBuilder disableCoord(boolean disableCoord) {
         return this;
     }
 
+    public boolean disableCoord() {
+        return this.disableCoord;
+    }
+
+    public boolean isTermsLookupQuery() {
+        return this.termsLookup != null;
+    }
+
+    public TermsQueryBuilder termsLookup(TermsLookupBuilder termsLookup) {
+        this.termsLookup = termsLookup;
+        return this;
+    }
+
+    public TermsLookupBuilder termsLookup() {
+        return this.termsLookup;
+    }
+
     /**
      * Sets the index name to lookup the terms from.
      */
     public TermsQueryBuilder lookupIndex(String lookupIndex) {
-        this.lookupIndex = lookupIndex;
+        if (lookupIndex != null) {
+            if (this.termsLookup == null) {
+                this.termsLookup = new TermsLookupBuilder();
+            }
+            this.termsLookup.index = lookupIndex;
+        }
         return this;
     }
 
+    public String lookupIndex() {
+        return this.termsLookup.index;
+    }
+
     /**
      * Sets the index type to lookup the terms from.
      */
     public TermsQueryBuilder lookupType(String lookupType) {
-        this.lookupType = lookupType;
-        return this;
+        if (lookupType != null) {
```

Comment:
true but then we also don't need to validate for those then. This is more to be consistent with what the parser already does.